### PR TITLE
[core] #346 spark job for suggestion index

### LINKED ISSUE #346: Suggestion API:
## Motivation

Suggestions API will let our clients to query with some imports and in response we will be able to give a closest and idea (based on ranking) suggestion to search with.

## Implementation details

Query will include a group of imports pretty much like we query importsmethods. But response will be a list of suggestions, each suggestion comprising of a group of FQCN as imports.

To implement this we need a spark job that can count transactions and store aggregated results like this.

Suggestion1: [< import1, import2. ...> `<count>`] 
Suggestion2: [< import2, import3. ...> `<count>`] 

On this index we will search using `should` and `sort` on `count`.

### PR DESCRIPTION
**Pull request by PiyushMishra**
fixes #346

### CODE REVIEW

File: bin/upload_to_es.sh [hunk 1]
```diff
@@ -25,6 +25,7 @@ curl -XDELETE 'http://localhost:9201/sourcefile/'
 curl -XDELETE 'http://localhost:9201/repository/'
 curl -XDELETE 'http://localhost:9201/importsmethods/'
 curl -XDELETE 'http://localhost:9201/statistics/'
+curl -XDELETE 'http://localhost:9201/statistics/'
```
Comment (ScrapCodes) [inline bin/upload_to_es.sh]: repeated line above.

File: core/src/main/scala/com/kodebeagle/spark/CreateIndexJobForSuggestion.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    // Create indexes for elastic search.
+    zipFileExtractedRDD.map { f =>
+      val (files, repo, packages, stats) = f
+      (repo, new JavaASTBasedIndexerForMethods().
+        generateTokensWithMethods(files.toMap, packages, repo), mapToSourceFiles(repo, files),
+        stats)
+    }.flatMap { case (_, importsMethods, _, _) => importsMethods.map(_.tokens.map(_.importName))
+      case _ => Seq()
+    }.groupBy(importSet => importSet).map {
```
Comment (ScrapCodes) [inline core/src/main/scala/com/kodebeagle/spark/CreateIndexJobForSuggestion.scala]: You can use reduceByKey.

File: core/src/main/scala/com/kodebeagle/spark/CreateIndexJobForSuggestion.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      case (importSetKey, importSet) => importSetKey -> importSet.size
+    }.sortBy(_._2).filter { case (a, b) => a.size > 1}.map
```
Comment (ScrapCodes) [inline core/src/main/scala/com/kodebeagle/spark/CreateIndexJobForSuggestion.scala]: Why are we sorting ?

File: core/src/main/scala/com/kodebeagle/spark/CreateIndexJobForSuggestion.scala (hunk 2 above)
Comment (PiyushMishra) [inline core/src/main/scala/com/kodebeagle/spark/CreateIndexJobForSuggestion.scala]: It's not a List of tuple, its a list of Set.

File: bin/upload_to_es.sh (hunk 1 above)
Comment (PiyushMishra) [inline bin/upload_to_es.sh]: Will correct it.

**Comment by ScrapCodes** [comment]
Is this updated ? Aren't we planning to keep both import name and import exact name.

**Comment by PiyushMishra** [comment]
Ya I have change in my local, will commit it.

**Pull request closed without merging**